pipedrive/client-php
# GRAL-2095 Add /callLogs endpoints

With this change it will be possible to make calls for all the /callLogs endpoints.

**How to test**
- Checkout this branch;
- Set up an app as instructed in the repo's readme, but add your local repo to compose and add this branch's contents as a dependency instead of pipedrive/pipedrive, or simply replace the contents of pipedrive/pipedrive in your app's vendor directory with this branch contents;
- Autoload a file with the code bellow to try it out, don't forget to replace the path to the autoload lib with one that works in your app, plus replace values for $apiToken with your Pipedrive API Token, $audioFilePath with the path to a .wav file you have available, plus $dealId with a deal ID you actually have in PD. This example will delete the file in the end, so comment the last call if you wish to see the log and play the audio in PD.
```php
<?php

require_once __DIR__.'/../vendor/autoload.php';

session_start();

// Client configuration
$apiToken = 'YOUR API TOKEN';
$audioFilePath = __DIR__.'/recording-file.wav';
$dealId = 1;

$client = new Pipedrive\Client(null, null, null, $apiToken); // First 3 parameters are for OAuth2

try {
    // Add a Call Log

    echo '<div><h2>Add a Call Log</h2><div>';
    $body = [
        'outcome' => 'connected',
        'to_phone_number' => '+372 53570010',
        'start_time' => '2021-07-23 07:00:00',
        'end_time'=> '2021-07-23 07:05:00',
        'deal_id'=> $dealId,
    ];
    $response = $client->getCallLogs()->addACallLog($body);
    echo '<pre>';
    var_dump($response);
    echo '</pre>';

    $callLogId = $response->data->id;
    echo 'Call log ID: ' . $callLogId;

    // Get all call logs
    echo '<div><h2>Get all call logs assigned to a particular user</h2><div>';
    $body = [
        'start' => 1,
    ];
    $response = $client->getCallLogs()->getAllCallLogsAssignedToAParticularUser($body);

    echo '<pre>';
    var_dump($response);
    echo '</pre>';

    // Get details of a call log
    echo '<div><h2>Get details of a call log</h2><div>';
    $response = $client->getCallLogs()->getDetailsOfACallLog($callLogId);

    echo '<pre>';
    var_dump($response);
    echo '</pre>';

    // Attach an audio file to the call log
    echo '<div><h2>Attach an audio file to the call log</h2><div>';
    $body = [
        'file' => $audioFilePath,
        'mime_type' => 'audio/wave',
    ];
    $response = $client->getCallLogs()->attachAnAudioFileToTheCallLog($callLogId, $body);

    echo '<pre>';
    var_dump($response);
    echo '</pre>';

    // Delete a call log
    echo '<div><h2>Delete a call log</h2><div>';
    $response = $client->getCallLogs()->deleteACallLog($callLogId);

    echo '<pre>';
    var_dump($response);
    echo '</pre>';

} catch (\Pipedrive\APIException $e) {
    echo $e;
}

```
